File: LeetCode/python3/1018.py
class Solution:
    def prefixesDivBy5(self, A: List[int]) -> List[bool]:
        num = 0
        ret = [False for _ in range(len(A))]
        for i in range(len(A)):
            num = (num * 2 + A[i]) % 5
            ret[i] = True if num == 0 else False
        return ret

File: LeetCode/cpp/1018.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int> &A) {
        int sum = 0;
        vector<bool> ret(A.size(), false);
        for (int i = 0; i < A.size(); ++i) {
            sum = (sum * 2 + A[i]) % 5;
            ret[i] = sum == 0 ? true : false;
        }
        return ret;
    }
};
